org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JDTUtils.java

```diff
@@ -583,6 +620,59 @@ public static IFile findFile(URI uri) {
 		}
 	}
 
+	public static IFolder findFolder(URI uri) {

```

Comment:
please refactor this method with `findFile`, there's room here to introduce a `findResources` method to avoid duplicate code.